src/components/button.tsx

```diff
@@ -0,0 +1,25 @@
+/* /components/button.tsx  */
+
+type ButtonType = 'button' | 'submit' | 'reset';
+interface Props {
+  children: React.ReactNode;
+  handleClick?: () => void;
+  styles: string;
+  type?: ButtonType;
+  title: string;
+}
+
+const Button = (props:Props) => {
+  return (
+      <button
+      onClick={props?.handleClick}
+      className={`${props.styles} px-3 py-2 rounded-lg font-medium border-2 border-transparent`}
+      type={props?.type}
+      title={props.title}
+      >
+          {props.children}
+      </button>
+  )
```

Comment:
A couple points here!

1. Since these props are directly tied to  `<button>` we should try to match the prop name with the actual prop we're going to pass into.
2. Optional props should have default values.
3. Interface name should be something meaningful like `ButtonProps`
4. Components should use named exports instead of default exports.

```tsx
interface ButtonProps {
  children: React.ReactNode;
  onClick?: () => void;
  className?: string;
  type?: ButtonType;
  title: string;
}

export const Button = ({
  children,
  onClick,
  className = '',
  type = 'button',
  title,
}: ButtonProps) => {
  return (
    <button
      onClick={onClick}
      className={`px-3 py-2 rounded-lg font-medium border-2 border-transparent ${className}`}
      type={type}
      title={title}
    >
      {children}
    </button>
  );
};

Reply:
Understood